api/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM node:latest AS dev
```

Comment:
Wait, this works? You can define two image build scripts in one Dockerfile? How do you select between them when running dev vs prod environments?

Reply:
Yeah, it's relatively new: https://docs.docker.com/develop/develop-images/multistage-build/

By default it builds all targets in the file but you can choose to build a specific target with the `--target` option, which is also supported by docker-compose.

One of the other neat things you can do with multistage builds is to install a bunch of build dependencies and do a build in one image, then copy just the thing that was built into a second image. I plan on doing this for the prod build of the app, so you'll see it in action soon.